### ISSUE TITLE
Microsoft.Azure.Devices.Client throwing ObjectDisposedException while using AMQP over websocket protocol

### ISSUE THREAD

**Issue by SwatantraYadav** [Azure/azure-iot-sdk-csharp#806]
**Description:**
Microsoft.Azure.Devices.Client library throws an unobserved ObjectDisposedException while Get or using an connection.

**Code:**
iotDeviceClient = DeviceClient.CreateFromConnectionString(deviceToken, Microsoft.Azure.Devices.Client.TransportType.Amqp_WebSocket_Only);

**Exception:**
Message:An exception occured in an unobserved Task.
Type: System.AggregateException
Message: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.
HResult: -2146233088

---- Inner Exception #1-1 ----

Type: System.ObjectDisposedException
Message: Cannot access a disposed object.
Object name: 'FaultTolerantAmqpObject`1'.
ObjectName: FaultTolerantAmqpObject`1
StackTrace:    at Microsoft.Azure.Devices.Client.Singleton`1.<GetOrCreateAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Devices.Client.IotHubConnection.<GetSessionAsync>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Devices.Client.IotHubConnection.<CreateReceivingLinkAsync>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Devices.Client.Transport.AmqpTransportHandler.<CreateDeviceBoundReceivingLinkAsync>d__66.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Devices.Client.FaultTolerantAmqpObject`1.<OnCreateAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Devices.Client.Singleton`1.<CreateValue>d__19.MoveNext()
Source: Microsoft.Azure.Devices.Client
HResult: -2146232798
Add to Search
OSVersion:Windows 10 Home  6.3.17763 x64

**Comment by CIPop** [comment]
@SwatantraYadav please fill in the template. This bug doesn't have all required information such as platform, .NET Runtime, SDK version.
Using defaults and the latest (non-preview) release the Singleton code causing the stack above shouldn't be instantiated. Are you using AMQP connection pooling and multiplexing?

**Comment by az-iot-builder-01** [comment]
@SwatantraYadav, @CIPop, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)

**Comment by SwatantraYadav** [comment]
@CIPop The library version and .Net Framework version are given below.
  <package id="Microsoft.Azure.Devices.Client" version="1.6.1" targetFramework="net46" />

I am not using Multiplexing. The protocol I am using is AMQP over web sockets.
Now i have taken the preview version and built 1.19 that i am trying to use. If that does not work then I will create a new issue.

**Comment by CIPop** [comment]
> Will the preview branch work with .NET Framework 4.6?

The `preview` branch should work with .NET Framework 4.6. If it doesn't please open a github issue.
`amqp_redesign_preview` is a development branch and may not support all runtimes yet.
